hardhat.config.ts

```diff
@@ -23,19 +23,19 @@ dotenv.config();
 // the following config is true.
 const compileZk = process.env.COMPILE_ZK === "true";
 
-const solcVersion = "0.8.17";
+const solcVersion = "0.8.18";
 const mnemonic = getMnemonic();
 
 // Compilation settings are overridden for large contracts to allow them to compile without going over the bytecode
 // limit.
 const LARGE_CONTRACT_COMPILER_SETTINGS = {
   version: solcVersion,
-  settings: { optimizer: { enabled: true, runs: 200 } },
+  settings: { optimizer: { enabled: true, runs: 200 }, viaIR: true },
```

Comment:
What does `viaIR` do?

Reply:
It essentially allows us to get around stack too deep errors. It modifies the compilation process to allow solidity to use an intermediate representation. Not sure of the details other than this is what the compiler recommends when you encounter stack too deep errors.

Reply:
oh wow didnt know about that, we should always have used that then?

Reply:
It's a new thing somewhere in the 0.8.X releases.

Reply:
There are a few instances where additional functions have been added to work around stack too deep issues. Would it be worth revisiting those now that this option is available?